modules/api/src/main/java/org/apache/ignite/table/TupleImpl.java

```diff
@@ -358,6 +360,17 @@ private void readObject(java.io.ObjectInputStream in) throws IOException, ClassN
         }
     }
 
+    /**
+     * Serializes an object. Required to be implemented in pair with the {@link #readObject(ObjectInputStream)} so that the
+     * {@code StructuredObjectMarshaller#fillStructuredObjectLayerFrom} will actually call the {@link #readObject(ObjectInputStream)}.
+     *
+     * @param out Output object stream.
+     * @throws IOException If failed.
+     */
+    private void writeObject(ObjectOutputStream out) throws IOException {
+        out.defaultWriteObject();
```

Comment:
This uses Java serialization, right? Can we use `MARSHALLED_TUPLE` mode for tuples instead?

Reply:
> Do you mean TupleWithSchemaMarshalling?
Yes. 
We should prefer formats that can be used by non-Java clients as well.

Reply:
What happens if I run the reproducer in the ticket using the client API?